### PR TITLE
Added stuff to DB and made API call to fetch the photos

### PR DESCRIPTION
**Pull request by hernandezvictoria**
## Description

- added some products to DB
- made API call to fetch photos
- display products (need to work on UI still)
- updated comments from last PR
- NOTE: limited to 500 API calls per month, so caching data from the API calls to limit the amount of calls

## Milestones
- API call integration

## Resources
- kudosboard logic

## Test Plan
video: 
https://www.loom.com/share/069b4641e2e54d818884f268da530193?sid=e520d8ba-cd4a-4a56-a4a3-fc67110ac21b

### CODE REVIEW

File: backend/data-processing/importProducts.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    let ingredientsArray = row.ingredients.split(',').map(i => i.trim());
+    ingredientsArray = ingredientsArray.map(i => i.toLowerCase());
+
+    const concernsArray = getConcernsArray(row.concerns);
+    const skinTypeArray = getSkinTypesArray(row.skin_type);
+
+    // Insert into database
+    try {
+      await prisma.productInfo.create({
```
Comment (ryanwproulx) [inline backend/data-processing/importProducts.js:64]: Make sure to test these edge cases with bad data

File: backend/prisma/migrations/20250626053742_added_more_product_info/migration.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+-- AlterTable
+ALTER TABLE "ProductInfo" ADD COLUMN     "price" TEXT,
+ADD COLUMN     "product_type" TEXT,
```
Comment (ryanwproulx) [inline backend/prisma/migrations/20250626053742_added_more_product_info/migration.sql:3]: For something like product_type here, where there is a set number of types, it is a good idea to have a separate table that matches a product_type_id to the product_type_name. Then in ProductInfo table you would have a product_type_id column matching to the other table. 

This is also a good idea for data consistency, ensuring that only valid product types are stored in the DB. It also saves storage space (especially if there were to ever be a lot of these stored) and reduces data duplication.

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -30,7 +30,10 @@ model ProductInfo{
   id Int @id @default(autoincrement()) //id in the database
   brand String
   name String
-  ingredients Json
+  image String?
+  product_type String?
+  price String?
+  ingredients Json?
   concerns Json?
   skin_type Json?
```
Comment (ryanwproulx) [inline backend/prisma/schema.prisma:38]: same as above if skin_type can have multiple options. Otherwise make it a (String?)

File: backend/routes/products.js [hunk 4]
```diff
@@ -23,5 +22,33 @@ router.get('/products', async (req, res) => {
 
 });
 
+// http://localhost:3000/change-product-image
```
Comment (ryanwproulx) [inline backend/routes/products.js:25]: How often and in what context will this get called? Just trying to wrap my head around it

File: frontend/meta-capstone/src/components/Quiz.jsx [hunk 5]
```diff
@@ -103,33 +102,21 @@ const Quiz = () => {
...
-                <label>{SkinTypes.OILY}</label><br></br>
-                <input type="checkbox" name="type" id={SkinTypes.DRY} value={SkinTypes.DRY}/>
-                <label>{SkinTypes.DRY}</label><br></br>
-                <input type="checkbox" name="type" id={SkinTypes.SENSITIVE} value={SkinTypes.SENSITIVE}/>
-                <label>{SkinTypes.SENSITIVE}</label><br></br>
-                <input type="checkbox" name="type" id={SkinTypes.COMBINATION} value={SkinTypes.COMBINATION}/>
-                <label>{SkinTypes.COMBINATION}</label><br></br>
-                <input type="checkbox" name="type" id={SkinTypes.NORMAL} value={SkinTypes.NORMAL}/>
-                <label>{SkinTypes.NORMAL}</label><br></br>
+                {Object.values(SkinTypes).map((skinType) => (
```
Comment (ryanwproulx) [inline frontend/meta-capstone/src/components/Quiz.jsx:105]: Nice change!

File: frontend/meta-capstone/src/components/home_components/Product.jsx [hunk 6]
```diff
@@ -1,15 +1,86 @@
...
+          'x-rapidapi-key': import.meta.env.VITE_API_KEY,
+          'x-rapidapi-host': 'real-time-sephora-api.p.rapidapi.com'
+        }
+      };
+
+      try {
+        const response = await fetch(url, options);
+        const result = await response.json();
+        const products_list = result.data.products;
+        const fetchedImage = products_list[0].heroImage;
```
Comment (ryanwproulx) [inline frontend/meta-capstone/src/components/home_components/Product.jsx:26]: add a check if products_list is empty

File: frontend/meta-capstone/src/components/home_components/ProductList.jsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    if(error){
+      return(<p>{error}</p>);
+    }
+
+    else{
+      return (
+        <div className="product-container">
+        {
+          data.map(obj => {
+            return(<Product
```
Comment (ryanwproulx) [inline frontend/meta-capstone/src/components/home_components/ProductList.jsx:43]: make sure to handle null cases for the nullable properties in the Product component

**Review by ryanwproulx — CHANGES_REQUESTED** [review]
Overall looking good! I would make those changes to the DB schema, but the rest of what I noticed were small and easy changes
